### DISCUSSION TITLE
Different phases at triple junctions

### DISCUSSION THREAD

**Discussion by prsdshiva** [neperfepx/neper#348] [Q&A] [answered]
I want to generate 3D grain tesselations containing different material phases at the triple junction. I should also be able to control the volume fraction of the second phase materials. (representative image attached here with)

Any thoughts on how to get this done?

Regards
Shiva 

![secondphase](https://user-images.githubusercontent.com/27774918/146190332-d592dfac-c848-411a-8b0c-57a068dc5033.png)

**Comment by smallrain1996** [comment]
Can you draw a diagram that I can't quite understand from your text what you want?

**Reply by prsdshiva** [reply]
I have updated the question with a representative image.

**Comment by rquey** [comment, marked as answer]
If the 2nd phase is at quadruple points (in 3D) or at triple points (in 2D) and made of smaller grains than for the 1st phase, then you can get such a structure by imposing a bimodal cell size distribution and assigning big cells (phase 1) and small cells (phase 2) to different *groups*:
```
neper -T -n from_morpho -morpho "diameq:lognormal(0.3,0.03)+0.5*lognormal(0.05,0.005),1-sphericity:lognormal(0.145,0.03)" -group "diameq>0.15?1:2" -morphooptiini "weight:radeq,coo:packing" -morphooptistop val=0.05 -o d
```

- `-morphooptiini "weight:radeq,coo:packing"` is not mandatory, but it speeds up the process.
- `-morphooptistop val=0.05` is to stop the optimization process early.

Visualize:
```
neper -V d.tess -datacellcol group -print img1 -showcell "group==2" -showedge all -print img2
```
![image](https://user-images.githubusercontent.com/2659376/148366188-e96336d5-3a47-460e-b3e6-7205853850f0.png)
![image](https://user-images.githubusercontent.com/2659376/148366202-106d3191-2ce4-4f6f-b95d-94db46ed83a0.png)

**Reply by Ava-22-neper** [reply]
@rquey 
Dear Prof,

It is possible in this case to choose larger sphericity for the green phase only?
